app/controllers/application_controller.rb

```diff
@@ -39,6 +45,19 @@ def not_found_error
 
   private
 
+  def store_user_location!
+    store_location_for(:user, request.fullpath)
+  end
+
+  # Its important that the location is NOT stored if:
+  # - The request method is not GET (non idempotent)
+  # - The request is handled by a Devise controller such as Devise::SessionsController as that could cause an
+  #    infinite redirect loop.
+  # - The request is an Ajax request as this can lead to very unexpected behaviour.
+  def storable_location?
```

Comment:
I wonder if we can add the check for the lesson page here? and maybe use the controller name and action of the request. Just incase `lessons` is in the full path for an unrelated page.

We should be able to do something like:
```
def storable_location?
    request.get? && is_navigational_format? && !devise_controller? && !request.xhr? && lesson_page?
 end

def lesson_page?
    controller_path == "lessons" && action_name == "show"
 end
```

Then the `after_sign_in_path_for` method could be simplified to:
```
def after_sign_in_path_for(resource)
    stored_location_for(resource) || dashboard_path
end
```

Reply:
Yeah, that's a better way to do it for sure.